spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/CompositeRouteLocator.java

```diff
@@ -69,6 +71,17 @@ public Route getMatchingRoute(String path) {
 	}
 
 	@Override
+	public Route getMatchingRoute(RequestContext requestContext) {
```

Comment:
I was thinking the actual `HttpServletRequest`, rather than a zuul context object. That can be accessed at any point since it's a static threadlocal bound instance.